### PR TITLE
Lodash upgrade

### PR DESCRIPTION
**Pull request by JoelMcKinnon**
Github security alert said we have a critical need to upgrade from v3.10.1 to 4.17.12 or later. Steps taken:
1. Change package.json and reinstall.
2. Note errors in console involve invalid methods
3. Review changelog and rename all methods in use that are straight one for one rename.
4. Review methods in use that have been modified in some way.
5. Install linter for lodash to identify other issues and fix.
6. Test for functionality, errors in console, run unit tests.

Opened separate issue for lodash import scope restrictions (identified by linter): https://github.com/nasa/openmct/issues/2884

Resolves: #2835

Author Checklist

- Changes address original issue? Yes
- Unit tests included and/or updated with changes? N/A
- Command line build passes? Yes
- Changes have been smoke-tested? Yes

### CODE REVIEW

**Comment by deeptailor** [comment]
failing tests, please update

**Comment by JoelMcKinnon** [comment]
Need some help smoke testing changes to TelemetryTableConfiguration.js line 103 and
SortedTableRowCollection.js line 96

File: src/plugins/plot/src/chart/MCTChartController.js [hunk 1]
```diff
@@ -150,7 +150,7 @@ function (
     MCTChartController.prototype.destroy = function () {
         this.isDestroyed = true;
         this.stopListening();
-        _.invoke(this.lines, 'destroy');
+        _.invokeMap(this.lines, 'destroy');
```
Comment (akhenry) [inline src/plugins/plot/src/chart/MCTChartController.js:153]: Let's re-implement natively.

File: src/api/telemetry/DefaultMetadataProvider.js [hunk 2]
```diff
@@ -85,7 +85,7 @@ define([
                             value: +e.value
                         };
                     }), 'e.value');
-                valueMetadata.values = _.pluck(valueMetadata.enumerations, 'value');
+                valueMetadata.values = _.map(valueMetadata.enumerations, 'value');
```
Comment (akhenry) [inline src/api/telemetry/DefaultMetadataProvider.js:88]: Let's re-implement natively.

File: src/api/telemetry/TelemetryMetadataManager.js [hunk 3]
```diff
@@ -57,7 +57,7 @@ define([
 
         if (valueMetadata.format === 'enum') {
             if (!valueMetadata.values) {
-                valueMetadata.values = _.pluck(valueMetadata.enumerations, 'value');
+                valueMetadata.values = _.map(valueMetadata.enumerations, 'value');
```
Comment (akhenry) [inline src/api/telemetry/TelemetryMetadataManager.js:60]: Let's reimplement natively.

File: src/plugins/plot/src/telemetry/StackedPlotController.js [hunk 4]
```diff
@@ -76,7 +76,7 @@ define([
                 if (childObj) {
                     var index = telemetryObjects.indexOf(childObj);
                     telemetryObjects.splice(index, 1);
-                    $scope.$broadcast('plot:tickWidth', _.max(tickWidthMap));
+                    $scope.$broadcast('plot:tickWidth', _.max(Object.values(tickWidthMap)));
```
Comment (akhenry) [inline src/plugins/plot/src/telemetry/StackedPlotController.js:79]: `Math.max()` does this, let's reimplement natively.

File: platform/telemetry/src/TelemetryCapability.js [hunk 5]
```diff
@@ -235,7 +235,7 @@ define(
             var defaultRange = metadata.valuesForHints(['range'])[0];
             defaultRange = defaultRange ? defaultRange.key : undefined;
 
-            var sourceMap = _.indexBy(metadata.values(), 'key');
+            var sourceMap = _.keyBy(metadata.values(), 'key');
```
Comment (akhenry) [inline platform/telemetry/src/TelemetryCapability.js:238]: Let's re-implement natively with map

File: platform/telemetry/src/TelemetryCapability.js [hunk 6]
```diff
@@ -300,7 +300,7 @@ define(
...
```
Comment (akhenry) [inline platform/telemetry/src/TelemetryCapability.js:303]: Let's re-implement natively with map

File: src/api/composition/DefaultCompositionProvider.js [hunk 7]
```diff
@@ -196,8 +196,8 @@ define([
      * @private
      */
     DefaultCompositionProvider.prototype.includes = function (parent, childId) {
-        return parent.composition.findIndex(composee =>
-            this.publicAPI.objects.areIdsEqual(composee, childId)) !== -1;
+        return parent.composition.some(composee =>
```
Comment (akhenry) [inline src/api/composition/DefaultCompositionProvider.js:199]: 👍

File: src/plugins/plot/src/chart/MCTChartController.js (hunk 1 above)
Comment (JoelMcKinnon) [reply src/plugins/plot/src/chart/MCTChartController.js]: Done

File: src/api/telemetry/DefaultMetadataProvider.js (hunk 2 above)
Comment (JoelMcKinnon) [reply src/api/telemetry/DefaultMetadataProvider.js]: Done

File: src/api/telemetry/TelemetryMetadataManager.js (hunk 3 above)
Comment (JoelMcKinnon) [reply src/api/telemetry/TelemetryMetadataManager.js]: Done

File: src/plugins/plot/src/telemetry/StackedPlotController.js (hunk 4 above)
Comment (JoelMcKinnon) [reply src/plugins/plot/src/telemetry/StackedPlotController.js]: Done

File: platform/telemetry/src/TelemetryCapability.js (hunk 5 above)
Comment (JoelMcKinnon) [reply platform/telemetry/src/TelemetryCapability.js]: Done, but not with map(). Used reduce().

File: platform/telemetry/src/TelemetryCapability.js (hunk 6 above)
Comment (JoelMcKinnon) [reply platform/telemetry/src/TelemetryCapability.js]: Done, but not with map(). Used reduce().

**Comment by JoelMcKinnon** [comment]
Installed and configured (but did not commit) an eslint plugin: https://github.com/you-dont-need/You-Dont-Need-Lodash-Underscore. After making the requested changes it still finds 18 problems (see below). I would like to take some time trying to fix some of these in a separate PR.

Screen Shot 2020-04-22 at 3 34 16 PM

File: platform/telemetry/src/TelemetryCapability.js (hunk 5 above)
Comment (akhenry) [reply platform/telemetry/src/TelemetryCapability.js]: Ah, yep, my bad, `reduce()` makes a lot more sense :)

**Comment by akhenry** [comment]
## Reviewer Checklist

* Changes appear to address issue? Y
* Appropriate unit tests included? N/A
* Code style and in-line documentation are appropriate? Y
* Commit messages meet standards? Y

**Pull request merged**